src/renderer/dx/DxFontRenderData.h

```diff
@@ -75,26 +75,29 @@ namespace Microsoft::Console::Render
         [[nodiscard]] static HRESULT STDMETHODCALLTYPE s_CalculateBoxEffect(IDWriteTextFormat* format, size_t widthPixels, IDWriteFontFace1* face, float fontScale, IBoxDrawingEffect** effect) noexcept;
 
     private:
-        void _BuildFontRenderData(const FontInfoDesired& desired, FontInfo& actual, const int dpi);
-        Microsoft::WRL::ComPtr<IDWriteTextFormat> _BuildTextFormat(const DxFontInfo fontInfo, const std::wstring_view localeName);
+        using FontAttributeMapKey = uint32_t;
 
-        ::Microsoft::WRL::ComPtr<IDWriteFactory1> _dwriteFactory;
+        // We use this to identify font variants with different attributes.
+        static FontAttributeMapKey _ToMapKey(DWRITE_FONT_WEIGHT weight, DWRITE_FONT_STYLE style, DWRITE_FONT_STRETCH stretch) noexcept {
+            return static_cast<FontAttributeMapKey>((weight << 16) | (style << 8) | stretch);

```

Comment:
Is `<< 8` enough? The max value of `DWRITE_FONT_WEIGHT` is 999.

Reply:
Yeah, since when we shift the weight upwards by 16 bits, a weight of 999 (`aaaa`), a style of 100 (`bb`) and stretch of 200 (`cc`) will look like this:
```
0x03E764C8
  aaaabbcc
```

I figured that hashing less data using FNV1a (which is a bytewise algorithm) in `std::hash` is less expensive than quickly shifting and ORing a couple integers.

Reply:
Thanks. I thought `DWRITE_FONT_STYLE` was also maxed at 999. Turned out it's not :).

Personally I prefer the shifting method used now, because it's clearer and intuitive. And also fast enough. But if people want the `std::hash` way I can change it back.